modules/gke-cluster/bin_auth.tf

```diff
@@ -0,0 +1,131 @@
+}
+
+resource "google_binary_authorization_policy" "policy" {
+  count    = "${var.enable_binary_authorization ? 1 : 0}"
+  provider = "google-beta.bin-auth"
+  project  = "${var.project_id}"
+
+  default_admission_rule {
+    evaluation_mode  = "${var.binary_authorization_evaluation_mode}"
+    enforcement_mode = "${var.binary_authorization_enforcement_mode}"
+  }
+
+  # Each `admission_whitelist_patterns` block supports only one `name_pattern`.
+  # To specify multiple patterns, we must specify multilple
+  # `admission_whitelist_patterns` blocks.
+  #
+  # This set of patterns provide some slots for module users to specify their
+  # custom patterns. It would be nice to use something like
+  # https://www.terraform.io/docs/configuration/expressions.html#dynamic-blocks
+  # but they are not available in Terraform 0.11 :(.
+
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_0}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_1}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_2}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_3}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_4}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_5}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_6}"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "${var.binary_authorization_admission_whitelist_pattern_7}"
+  }
+
+  # This set of patterns is for Helm.
+
+  admission_whitelist_patterns {
+    name_pattern = "gcr.io/kubernetes-helm/*"
+  }
+
+  # This set of patterns covers "system" images that are included in neither
+  # of Google's provided lists (short or full; see below).
+
+  admission_whitelist_patterns {
+    name_pattern = "gcr.io/gke-release/istio/*"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "gcr.io/istio-release/*"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "gcr.io/projectcalico-org/*"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "k8s.gcr.io/fluentd-gcp-scaler:*"
+  }
+  admission_whitelist_patterns {
+    name_pattern = "quay.io/prometheus/prometheus*"
+  }
+
+  # This set of patterns covers Google-provided system images. It would be nice
+  # to use the `globalPolicyEvaluationMode` field in the API
+  # (https://cloud.google.com/binary-authorization/docs/reference/rest/v1/Policy)
+  # but the google provider doesn't support it :(.
+  #
+  # The GENERATED section is manually generated by this process:
+  #
+  # 1. Collect list of Google-provided system images, e.g. the short list at
```

Comment:
I would remove the comment from this point below and just reference the GCP doc listing those 4 patterns below (https://cloud.google.com/binary-authorization/docs/configuring-policy-cli#global_policy_evaluation_mode). Reading & following the comment is longer than copying those 4 lines and the comment as a whole distracts from the actual reason - why those 4 patterns.